usaspending_api/search/delta_models/award_search.py

```diff
@@ -474,7 +482,64 @@
         GROUPED_BY_DEFC.award_id IS NOT NULL
     GROUP BY
         GROUPED_BY_DEFC.award_id
-) DEFC on DEFC.award_id = awards.id
+) COVID_DEFC on COVID_DEFC.award_id = awards.id
+-- Infrastructure Investment and Jobs Act (IIJA) spending
```

Comment:
Everything looks really good. But I have lots to suggest here. Rather than following suit of the SQL pattern, there is probably an opportunity that needs to be explored on whether we can create e.g. a CTE that is built _ONCE_ that aggregates costs by `defc.code` or by `defc.code, defc.group_name`. Then the aggregations across ALL records only needs to be done once, and you could (maybe?) pick out which group of agg'd values you're interested in by joining/querying the CTE.

Doing in this way will work, but you're paying the cost of looking at and aggregating these records twice, or nearly twice (maybe some efficiencies by limiting the aggregated rows by group -- which you can still do, but it would be limited to all groups you want to agg (currently 2), not just 1 group).

<img width="1015" alt="image" src="https://user-images.githubusercontent.com/14953413/223559208-8d50ae68-103c-46d2-8672-46fc1be4bad0.png">

Also, thank goodness for Spark. Something like this added to a Postgres materialized view SQL might have added hours to its runtime. You should still run this up in Databricks to get a sense of how much time it adds before/after.

Reply:
If this becomes impossible to create a generic grouping a CTE because of too many factors (group_name, date filtering, etc.) maybe just leave it as you have it.

Reply:
After some testing, converting this to use a CTE instead of the LEFT OUTER JOINs doesn't have much of an affect on performance. Either version of the INSERT SQL will complete in 27 to 25 minutes.